### PR TITLE
[TEAM2-499] Currency Select Modal Crosschain Search

### PR DESCRIPTION
**Pull request by derHowie**
## What changed (plus any additional context for devs)
Crosschain searches now provide a `fromChainId` query param. This filters the resulting list based on Socket's available routes.

## Screen recordings / screenshots
[This](https://recordit.co/BpFFLcY2xI) is an example of a crosschain search for 'aave'. Aave is a good test case because it has a lot of token derivates, many of which have low liquidity and are not supported by Socket. You'll notice in the video that L2 search UI contain mainnet's results for tokens outside of `verifiedAssets`. This is not related to my changes, but rather an existing issue with the feature branch we will need to address. I addressed it [here](https://github.com/rainbow-me/rainbow/pull/4221/files#r982904122).

## What to test
We should ensure that search functionality and the swap modals themselves still behave as they have in the past. Helping to verify that tokens not supported by Socket are not returned for a given chain combination will also help, however there are numerous cases to test and it would be impractical to do this exhaustively.

The filtering mechanism used is based on lists generated by [this ](https://api.socket.tech/v2/swagger/#/Token%20Lists/TokenListController_getToTokenList)endpoint.

When verifying whether a token should be included in a search, you can leverage the following request:
```
curl --location --request GET 'https://api.socket.tech/v2/token-lists/to-token-list?fromChainId=10&toChainId=1' --header 'API-KEY: 645b2c8c-5825-4930-baf3-d9b997fcd88c'
```
This curl example will return tokens supported by Socket when swapping from optimism (10) to mainnet (1).

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/hooks/useSwapCurrencyList.ts [hunk 1]
```diff
@@ -94,6 +106,20 @@ const useSwapCurrencyList = (
   const [lowLiquidityAssets, setLowLiquidityAssets] = useState<RT[]>([]);
   const [verifiedAssets, setVerifiedAssets] = useState<RT[]>([]);
 
+  const { inputCurrency } = useSwapCurrencies();
+  const crosschainSwapsEnabled = useExperimentalFlag(CROSSCHAIN_SWAPS);
+  const inputChainId = useMemo(
+    () =>
+      inputCurrency?.network &&
+      ethereumUtils.getChainIdFromNetwork(inputCurrency?.network as Network),
```
Comment (estebanmino) [inline src/hooks/useSwapCurrencyList.ts:114]: oh i think some tokens don't have the network param and we should use type instead

i believe there's a method called ethereumUtils.getChainIdFromType to do the same

File: src/hooks/useSwapCurrencyList.ts [hunk 2]
```diff
@@ -409,7 +462,7 @@ const useSwapCurrencyList = (
             data: curatedAssets,
             key: 'curated',
             title: tokenSectionTypes.verifiedTokenSection,
-            useGradientText: IS_TESTING === 'true' ? false : true,
+            useGradientText: IS_TESTING !== 'true',
```
Comment (estebanmino) [inline src/hooks/useSwapCurrencyList.ts:465]: i believe you can use IS_TEST now

File: src/hooks/useSwapCurrencyList.ts [hunk 3]
```diff
@@ -94,6 +106,20 @@ const useSwapCurrencyList = (
...
+  const { inputCurrency } = useSwapCurrencies();
+  const crosschainSwapsEnabled = useExperimentalFlag(CROSSCHAIN_SWAPS);
+  const inputChainId = useMemo(
+    () =>
+      inputCurrency?.network &&
+      ethereumUtils.getChainIdFromNetwork(inputCurrency?.network as Network),
+    [inputCurrency?.network]
+  );
+  const isCrosschainSearch = useMemo(() => {
+    if (inputChainId && inputChainId !== chainId && crosschainSwapsEnabled) {
```
Comment (estebanmino) [inline src/hooks/useSwapCurrencyList.ts:118]: this is just a naming comment. it took me a second to know where `chainId` was coming from and figured out is coming from currency select list network, not input nor output which makes sense, what do you think of changing the name internally to something like `searchChainId` or something? to make it clear that this value is for searching and not coming from tokens as i thought at the beginning

Comment (estebanmino) [reply src/hooks/useSwapCurrencyList.ts:118]: for ex. when i saw `fromChainId` i immediately assumed it was coming from the input

**Comment by estebanmino** [comment]
seems like seaching verified section is working nicely but the unverified / low liquidity sections persist mainnet results https://www.loom.com/share/252f7a1e0fb54756ac0f6c9ceebe3763

File: src/hooks/useSwapCurrencyList.ts (hunk 1 above)
Comment (estebanmino) [reply src/hooks/useSwapCurrencyList.ts:114]: cross commenting prs https://github.com/rainbow-me/rainbow/pull/4221/files#r982967221

File: src/hooks/useSwapCurrencyList.ts (hunk 2 above)
Comment (derHowie) [reply src/hooks/useSwapCurrencyList.ts]: Alright. I think this diff was generated by the formatter, but I changed these lines to use IS_TEST instead.

File: src/hooks/useSwapCurrencyList.ts (hunk 1 above)
Comment (derHowie) [reply src/hooks/useSwapCurrencyList.ts:114]: based on the cross comment, are you fine with leaving this as is? Just want to confirm.

**Review by estebanmino — APPROVED** [review]
🚢

File: src/hooks/useSwapCurrencyList.ts [hunk 4]
```diff
@@ -49,11 +52,13 @@ const abcSort = (list: any[], key?: string) => {
...
 };
 
-const searchCurrencyList = async (
-  searchList: RT[] | TokenSearchTokenListId,
-  query: string,
-  chainId: number
-) => {
+const searchCurrencyList = async (searchParams: {
+  chainId: number;
+  fromChainId?: number | '';
```
Comment (sammdec) [inline src/hooks/useSwapCurrencyList.ts:57]: should we use `string` type here rather than string literal

**Review by sammdec — APPROVED** [review]
LGTM 🚢

**Pull request merged**